tools/autograd/derivatives.yaml

```diff
@@ -2216,6 +2216,18 @@
   # self_is_result is always false here since double backward call is an out-of-place call, self is input itself
   grad_output: leaky_relu_backward(grad, self, negative_slope, false)
   self: zeros_like(grad)
+  # leaky_relu_backward(grad_output, self, negative_slope, false)
+  # computes grad_output * at::where(self_p > 0, 1, negative_slope)
+  # so the jvp formula is the following:
+  # grad_output_t * at::where(self_p > 0, self_p.new_ones([]), negative_slope);
+  #
+  # leaky_relu_backward(grad_output, result, negative_slope, true)
+  # computes grad_output * at::where(result > 0, 1, negative_slope)
+  # so the jvp formula is the following:
+  # grad_output_t * at::where(result_p > 0, result_p.new_ones([]), negative_slope);
+  #
+  # Combined together that results in the following optimized kernel:
+  result: leaky_relu_backward(grad_output_t, self_p, negative_slope, self_is_result)
```

Comment:
Any idea what self_is_result does here? It looks like were are computing the same thing.

Reply:
Ah good question. 

If `self_is_result` is True, and `negative_slope` is negative, then we have a problem. There's a check for that in https://github.com/pytorch/pytorch/blob/10023eeb65d7bc6c71f49bf0ce7eaf5b58d52fd7/aten/src/ATen/native/Activation.cpp#L121-L126 already: if the self_is_result is True and negative_slope is negative, then it raises an error.

I will beef up the comment to mention this